### ISSUE TITLE
Improvement for Pushed Assets checking in Push Publishing

### ISSUE THREAD

**Issue by dsilvam** [dotCMS/core#8579]
There is data stored in the publishing_pushed_assets table that is not needed for determining if an asset needs to be pushed or not. 

Currently the table looks like this:

```
create table publishing_pushed_assets(
    bundle_id varchar(36) NOT NULL,
    asset_id varchar(36) NOT NULL,
    asset_type varchar(255) NOT NULL,
    push_date DATETIME,
    environment_id varchar(36) NOT NULL
);

CREATE INDEX idx_pushed_assets_1 ON publishing_pushed_assets (bundle_id);
CREATE INDEX idx_pushed_assets_2 ON publishing_pushed_assets (environment_id);
CREATE INDEX idx_pushed_assets_3 ON publishing_pushed_assets (asset_id, environment_id);
```

Current behaviour is that we are repeating data in that table for every different bundle. So if N different bundles send the same asset, it will be inserted N times. This makes having to iterate through the results to determine if an asset needs to be re-pushed and makes the table huge if using push publishing on a regular basis, which ends up slowing down the process in a significant manner. 

What we need is to have a unique constraint for (asset_id, environment_id), so that way we can check if a particular asset was already sent to a particular environment, and if that is the case, check the push_date to see if a re-push is needed. Doing that will be a query like 'select \* from publishing_pushed_assets where asset_id = {assetId} and environment_id = {environmentId} which will return only one result and the query will take advantage of the index on the unique constraint. This traduces in speeding things up significantly when determining if an asset needs to be re-pushed. 

So to summarise changes needed:
- 'bundle_id' is not need and can be removed
- 'asset_type' is not need and can be removed
- unique constraint need to be created for (asset_id, environment_id)
- code needs to change to support new behaviour
